### PR TITLE
[FIX] documents: fix traceback caused by sample data

### PR DESCRIPTION
**Pull request by std-odoo**
Bug
===
When we show the sample data of documents, it will choose a random value for the selection field. But document use a special option to make the browser generate the thumbnail of PDFs and then save it on the record. So, without this rule, it will try to update the record if "client_generated" has been chosen.

Task-3697924

### CODE REVIEW

**Comment by std-odoo** [comment]
@Polymorphe57 Hi :) 

Do you think 2 fields is still ok to hardcode their names ?

**Comment by aab-odoo** [comment]
Hello @std-odoo 

I don't know what @Polymorphe57 thinks, but I think it's not ok. The first exception was a mail field, which is available on a lot of models (it's kind of a framework field). And we weren't proud of that. Yours isn't framework at all, it's a specific field of a specific model. So I'd say it's time to introduce the "clean" solution.

Note that there's a chance that sample data will be generated server side in the future, so if this ever happens, the override might become more natural/easier

**Comment by std-odoo** [comment]
@aab-odoo It's not really the framework, it's mail (which is not necessary installed if web is installed), but I get your point

> So I'd say it's time to introduce the "clean" solution.

Ok, I'll check :)

File: addons/mail/__manifest__.py [hunk 1]
```diff
@@ -128,6 +128,7 @@
             'mail/static/src/**/primary_variables.scss',
         ],
         'web.assets_backend': [
+            'mail/static/src/model/**/*',
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/mail/__manifest__.py:131]: Or maybe not, it follows structure of module from which it is overridden after all.

[the remark this answers is not in the document]
Comment (std-odoo) [reply addons/mail/__manifest__.py:131]: yes, if I move it, I should move the file in /web too :/

[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/mail/__manifest__.py:131]: Yes I see, it's just that `/web` is a big mix of web client (backend assets) and actual framework stuff. Having such specific code at the root feels very off.

Whereas we have to manage many different bundles for each of backend, portal, external livechat, discuss public page, ... So we need a much more refined structure.

File: addons/mail/static/src/model/sample_server.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/* @odoo-module */
+
+import { SampleServer } from "@web/model/sample_server";
+import { patch } from "@web/core/utils/patch";
+
+patch(SampleServer.prototype, {
+    _getRandomSelectionValue(modelName, field) {
```
Comment (awa-odoo) [inline addons/mail/static/src/model/sample_server.js:7]: Would keep the small docstring explanation part that mentions we don't want to show a bunch of warnings in the list.

File: addons/mail/__manifest__.py [hunk 3]
```diff
@@ -128,6 +128,7 @@
             'mail/static/src/**/primary_variables.scss',
```
Comment (awa-odoo) [inline addons/mail/__manifest__.py:128]: Commit message could mention that we add a hook method inside the "SampleServer" as well to be able to override the fields for which we want to return false for selections.

Currently it's very centered on the changes in mail.

The header could also mention the "web" module.

**Review by awa-odoo — COMMENTED** [review]
Hello @std-odoo 

A couple light comments otherwise LGTM.

Side-note: you don't have to manually add references towards the enterprise PR in the commit message and vice-versa, the mergebot does it for you.

Thanks & cheers!

**Review by std-odoo — COMMENTED** [review]
@awa-odoo Hi :) Change done 

Cheers

**Comment by awa-odoo** [comment]
@robodoo r+

**Pull request closed without merging**